### DISCUSSION TITLE
CodeMaid is dead

### DISCUSSION THREAD

**Discussion by icodeintx** [codecadwallader/codemaid#835] [General]
[deleted]

**Comment by codecadwallader** [comment]
This invoked a lot of negative emotions and I did not appreciate it.

**Reply by icodeintx** [reply]
My apologies Steve, my comment wasn't meant to be negative.  I was referring to lack of activity.  The product itself is 👍

**Reply by ghost** [reply]
This tool is a great helper that many have used for years. Please ignore negative comments.

* I would like to see support added for VS Code (if I can help with that, please contact me)

**Comment by CodeChief** [comment]
Just wondering, **constructively**, as the last release was in 2021, and CodeMaid started crashing after the most recent VS2022 update...

If this tool were to be made "up to date" and aligned with other refactoring and formatting tools, I would expect to see:

1. Shift of all settings to EditorConfig, mapped to standard settings when present, including replacement of custom settings dialog with VS-integrated settings.
3. Removal of features built-in to latest VS versions (including ditching support for older VS versions).
4. New releases every 1-6 months max, else it's no surprise if people consider it "dead".

The direction of travel, in my opinion, is that MS appear to be **finally** enabling more and more efficiency and comfort features (analysis, refactoring, formatting, etc...) in the core product. Traditionally, I note over the 2 decades I've been using Visual Studio, they appeared to support extension writers by not implementing features which would be trivial for them but they seemed to avoid on purpose. 

Tools like CodeMaid and ReSharper polyfil an ever decreasing gap. It would be realistic, especially if not so much time is available to make new releases, that this product was at least brought into conform with current standards (EditorConfig for example) and slimmed down, potentially getting to the point where it is no longer necessary or so easy to maintain with just a few viable features on top of the standard set.

We could guess if MS will migrate to VS code or keep the too-heavily Windows tied VS monster alive, as cross-platform and online is also a clear direction of travel. I wonder if there are any parallels to implementing a formatting extension for VS Code and VS, or if that makes sense as there could be differences in the built-in features.

The clear market demand, not currently filled by MS, is to have proactive formatting to standards both in VS and VS code.

Do you think realistically that this project is able to get there? I guess it needs a kick-off of a major design discussion on a new branch, or is something like this already in the works?

**Reply by a1ex9788** [reply]
@codecadwallader, can you explain your intentions?
